### PR TITLE
testing: add autosave before each test run

### LINKED ISSUE #139125: Add option to auto save when running tests
I have autosave enabled for changing focus. When I use the new testing framework, whether with the shortcuts (like `ctrl+;+t`) or the buttons in the testing panel, the focus doesn't change and thus tests are run on the old, undesired code. An option to autosave whenever running a testing command would be nice.

### PR DESCRIPTION
**Pull request by Kalmaegi**
Save the current project file before each test run so that the code logic is up to date when the test is run

This PR fixes #139125

### CODE REVIEW

**Comment by connor4312** [comment]
We should make this configurable, and do it like how debug does. For example: https://github.com/microsoft/vscode/blob/cd27cdc4e395d84ebbe2b8e4410735a19219d9a7/src/vs/workbench/contrib/debug/common/debugUtils.ts#L312-L325

**Comment by slidoooor** [comment]
> 我们应该让它可配置，并像调试那样做。例如：
> 
> https://github.com/microsoft/vscode/blob/cd27cdc4e395d84ebbe2b8e4410735a19219d9a7/src/vs/workbench/contrib/debug/common/debugUtils.ts#L312-L325
> 
> 此外，这可能只触发通过 UI 触发的运行，而不是 API。检查示例：
> 
> https://github.com/microsoft/vscode/blob/e80a49a73482a35d2568496b25dae7c9a6cc9cf3/src/vs/workbench/contrib/testing/browser/testingProgressUiService.ts#L61-L63

ok, this is indeed better

**Comment by slidoooor** [comment]
> We should make this configurable, and do it like how debug does. For example:

Okay！

**Comment by slidoooor** [comment]
@connor4312 Do you think we need to consider untitled file? running test is more frequent than starting debug, I was worried about the impact on the time

**Comment by connor4312** [comment]
I don't think we need to deal with untitled files. It's unlikely that test runners will discover tests in untitled files.

**Comment by slidoooor** [comment]
> I don't think we need to deal with untitled files. It's unlikely that test runners will discover tests in untitled files.

Yes, this is somewhat unnecessary

File: src/vs/workbench/contrib/debug/browser/debug.contribution.ts [hunk 1]
```diff
@@ -532,6 +532,16 @@ configurationRegistry.registerConfiguration({
 			default: 'allEditorsInActiveGroup',
 			scope: ConfigurationScope.LANGUAGE_OVERRIDABLE
 		},
+		'debug.saveBeforeTest': {
```
Comment (connor4312) [inline src/vs/workbench/contrib/debug/browser/debug.contribution.ts:535]: This should be in the test settings, not the debug settings

Comment (slidoooor) [reply src/vs/workbench/contrib/debug/browser/debug.contribution.ts:535]: Thanks for the reminder

Comment (slidoooor) [reply src/vs/workbench/contrib/debug/browser/debug.contribution.ts:535]: Thanks for the reminder!

Comment (slidoooor) [reply src/vs/workbench/contrib/debug/browser/debug.contribution.ts]: I tweaked it a bit and used the bool type，which seems to be sufficient for now

**Review by connor4312 — APPROVED** [review]

**Comment by connor4312** [comment]
Looks good, thanks! This'll go out in the release next week

**Comment by slidoooor** [comment]
> 看起来不错，谢谢！这将在下周发布

我最近正在学习和测试有关的工作，能帮倒忙我真的很开心！

**Comment by slidoooor** [comment]
> 看起来不错，谢谢！这将在下周发布

I'm learning to work with testing these days, and I'm really happy to help out!

**Comment by slidoooor** [comment]
> Looks good, thanks! This'll go out in the release next week

I'm learning to work with testing these days, and I'm really glad I could help!

**Pull request merged**